### PR TITLE
first cut and removing openGl dependency in shared

### PR DESCRIPTION
**Pull request by ZappoMan**

### CODE REVIEW

File: libraries/shared/CMakeLists.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@ set(TARGET_NAME shared)
 
 # use setup_hifi_library macro to setup our project and link appropriate Qt modules
 # TODO: there isn't really a good reason to have Script linked here - let's get what is requiring it out (RegisteredMetaTypes.cpp)
-setup_hifi_library(Gui Network OpenGL Script Widgets)
+setup_hifi_library(Gui Network Script Widgets)
 
 setup_hifi_opengl()
```
Comment (birarda) [inline libraries/shared/CMakeLists.txt]: guessing we don't want this either

Comment (birarda) [inline libraries/shared/CMakeLists.txt]: are boostconfig and oglplus links also required for windows below?

Comment (ZappoMan) [inline libraries/shared/CMakeLists.txt]: fixing

**Comment by hifi-gustavo** [comment]
Build console output for commit: 673e1ee0cc2fb4e1cd223ac1e287fc9ab11cb2e4 available at: https://gist.github.com/7c018af0dcf2ad56c35e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5157.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5157.dmg

File: libraries/render-utils/CMakeLists.txt [hunk 2]
```diff
@@ -1,27 +1,37 @@
-set(TARGET_NAME render-utils)
```
Comment (ZappoMan) [inline libraries/render-utils/CMakeLists.txt]: note: these changes are mostly just line ending changes.

File: libraries/render-utils/CMakeLists.txt [hunk 3]
```diff
@@ -1,27 +1,37 @@
...
+
+AUTOSCRIBE_SHADER_LIB(gpu model render)
+
+# pull in the resources.qrc file
+qt5_add_resources(QT_RESOURCES_FILE "${CMAKE_CURRENT_SOURCE_DIR}/res/fonts/fonts.qrc")
+
+# use setup_hifi_library macro to setup our project and link appropriate Qt modules
+setup_hifi_library(Widgets OpenGL Network Qml Quick Script)
+
+setup_hifi_opengl()
```
Comment (ZappoMan) [inline libraries/render-utils/CMakeLists.txt]: this is new

File: libraries/render-utils/CMakeLists.txt [hunk 4]
```diff
@@ -1,27 +1,37 @@
...
+    # try to find the Nsight package and add it to the build if we find it
+    find_package(NSIGHT)
+    if (NSIGHT_FOUND)
+      include_directories(${NSIGHT_INCLUDE_DIRS})
+      add_definitions(-DNSIGHT_FOUND)
+      target_link_libraries(${TARGET_NAME} "${NSIGHT_LIBRARIES}")
+    endif ()
+  endif()
+
+  add_dependency_external_projects(boostconfig)
```
Comment (ZappoMan) [inline libraries/render-utils/CMakeLists.txt]: from here line 28 to line 34 is also new

**Comment by hifi-gustavo** [comment]
Build console output for commit: 383a2c3d639746a561a36fa417f3431e3b0761aa available at: https://gist.github.com/7c018af0dcf2ad56c35e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5157.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5157.dmg

**Comment by ZappoMan** [comment]
test this please

**Comment by jherico** [comment]
Weird build error on Ubuntu

```
In file included from /mnt/ebs1/jenkins/jobs/hifi-pull-request-ubuntu/workspace/build/libraries/avatars/avatars_automoc.cpp:2:0:
/mnt/ebs1/jenkins/jobs/hifi-pull-request-ubuntu/workspace/build/libraries/avatars/moc_AvatarData.cpp:330:11: error: ‘glm::AvatarData’ has not been declared
 void glm::AvatarData::qt_static_metacall(QObject *_o, QMetaObject::Call _c, int _id, void **_a)
```

**Comment by hifi-gustavo** [comment]
Build console output for commit: 383a2c3d639746a561a36fa417f3431e3b0761aa available at: https://gist.github.com/7c018af0dcf2ad56c35e<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5157.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5157.dmg

**Pull request merged**